client/state/ui/guided-tours/contexts.js

```diff
@@ -19,15 +20,20 @@ export const isPreviewNotShowing = state =>
 export const isPreviewShowing = state =>
 	isPreviewShowingSelector( state );
 
+const timeSinceUserRegistration = state => {
+	const user = getCurrentUser( state );
+	return user ? ( Date.now() - Date.parse( user.date ) ) : false;
+};
+
 const WEEK_IN_MILLISECONDS = 7 * 1000 * 3600 * 24;
 export const isNewUser = state => {
-	const user = getCurrentUser( state );
-	if ( ! user ) {
-		return false;
-	}
+	const userAge = timeSinceUserRegistration( state );
+	return ! isNaN( userAge ) ? userAge <= WEEK_IN_MILLISECONDS : false;
+};
 
-	const creation = Date.parse( user.date );
-	return ( Date.now() - creation ) <= WEEK_IN_MILLISECONDS;
+export const isUserOlderThan = age => state => {
+	const userAge = timeSinceUserRegistration( state );
+	return ! isNaN( userAge ) ? userAge >= age : false;
```

Comment:
Again, if `timeSinceUserRegistration` could only return either a number, or false if anything goes wrong, the handling here would be nicer:

```js
return userAge ? userAge >= age : false;
// or even
return userAge && userAge >= age;
```